src/ddmd/mars.d

```diff
@@ -849,6 +850,31 @@ private int tryMain(size_t argc, const(char)** argv)
             printf("%.*s", cast(int)ob.offset, ob.data);
     }
 
+    if(global.params.compileimports && global.params.link)
+    {
+        auto importedModulesToCompile = Modules();
+        fprintf(global.stdmsg, "compileimports: start\n");
+        for (size_t i = 0; i < Module.amodules.dim; i++)
+        {
+            auto m = Module.amodules[i];
+            if(m.isRoot)
+            {
+                continue;
+            }
+            if(m.inALibrary(libmodules))
+            {
+                //fprintf(global.stdmsg, "not compiling \"%s\" because it is in a library\n", m.toChars());
+                continue;
+            }
+            fprintf(global.stdmsg, "compileimports: compiling \"%s\" (file \"%s\" importedFrom \"%s\")\n",
+                m.toChars(), m.srcfile.toChars, m.importedFrom.toChars());
```

Comment:
For Phobos the coding style is to avoid unnecessary parentheses, for DMD there isn't a style guide yet, but it's probably a good idea to be consistent within the same line